### PR TITLE
NIAD-1670: Concatenate DOS service id with  ODS code

### PR DESCRIPTION
**Pull request by mrogowska**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java [hunk 1]
```diff
@@ -23,8 +24,22 @@ public ItkReportHeader getHeaderValues(Element headerElement) {
 
     private void setAddressList(ItkReportHeader header, Element headerElement) {
         List<Element> elements = getElements(headerElement, ITK_ADDRESS_NODE);
-        List<String> addresses = elements.stream().map(it -> it.attributeValue("uri")).collect(toList());
-        header.setAddressList(addresses);
+        Optional<String> dosServiceId = elements.stream()
+            .filter(it -> it.attribute("type") != null)
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java:28]: could we check if type is equal to "2.16.840.1.113883.2.1.3.2.4.18.44"?

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java [hunk 2]
```diff
@@ -23,8 +24,22 @@ public ItkReportHeader getHeaderValues(Element headerElement) {
...
+            .map(it -> it.attributeValue("uri"))
+            .findFirst();
+
+        List<String> addressList = elements.stream()
+            .filter(it -> it.attribute("type") == null
+                && it.attributeValue("uri").contains("ods")
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java:34]: I'd replace contains("ods") with startsWith("urn:nhs-uk:addressing:ods:")

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java [hunk 3]
```diff
@@ -23,8 +24,22 @@ public ItkReportHeader getHeaderValues(Element headerElement) {
...
+            )
+            .map(it -> dosServiceId.isPresent()
+                ? it.attributeValue("uri") + ":DOSServiceID:" + dosServiceId.get()
+                : it.attributeValue("uri")
+            )
+            .collect(toList());
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java:40]: I'm not sure if ITK message can contain DOS ID only (no ODS code). This code assumes ODS is always there, I think we should treat it as optional.

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java (hunk 1 above)
Comment (mrogowska) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java]: Done.

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java (hunk 2 above)
Comment (mrogowska) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java]: Done.

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java (hunk 3 above)
Comment (mrogowska) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java]: Changed to allow for ODS + DOS ID, only ODS and only DOS ID.

**Review by ssokolowskisebastian — APPROVED** [review]

**Pull request merged**